dex/testing/loadbot/loadbot.go

```diff
@@ -0,0 +1,440 @@
+	dextestDir = filepath.Join(usr.HomeDir, "dextest")
+	botDir     = filepath.Join(dextestDir, "loadbot")
+
+	defaultRegFee uint64 = 1e8
+	ctx, quit            = context.WithCancel(context.Background())
+
+	alphaAddrDCR, betaAddrDCR, alphaAddrBTC, betaAddrBTC string
+	alphaCfgDCR, betaCfgDCR,
+	alphaCfgBTC, betaCfgBTC map[string]string
+
+	dcrAssetCfg, btcAssetCfg   *dexsrv.AssetConf
+	orderCounter, matchCounter uint32
+	epochDuration              uint64
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+// rpcAddr is the RPC address needed for creation of a wallet connected to the
+// specified asset node. For Decred, this is the dcrwallet 'rpclisten'
+// configuration parameter.For Bitcoin, its the 'rpcport' parameter.
+func rpcAddr(symbol, node string) string {
+	switch symbol {
+	case dcr:
+		switch node {
+		case alpha:
+			return alphaCfgDCR["rpclisten"]
+		case beta:
+			return betaCfgDCR["rpclisten"]
+		}
+	case btc:
+		switch node {
+		case alpha:
+			return alphaCfgBTC["rpcport"]
+		case beta:
+			return betaCfgBTC["rpcport"]
+		}
+	}
+	panic("rpcAddr: unknown symbol-node combo " + symbol + "-" + node)
+}
+
+// returnAddress is an address for the specified node's wallet. returnAddress
+// is used when a wallet accumulates more than the max allowed for some asset.
+//
+func returnAddress(symbol, node string) string {
+	switch symbol {
+	case dcr:
+		switch node {
+		case alpha:
+			return alphaAddrDCR
+		case beta:
+			return betaAddrDCR
+		}
+	case btc:
+		switch node {
+		case alpha:
+			return alphaAddrBTC
+		case beta:
+			return betaAddrBTC
+		}
+	}
+	panic("returnAddress: unknown symbol-node combo " + symbol + "-" + node)
+}
+
+// mineAlpha will mine a single block on the alpha node of the asset indicated
+// by symbol.
+func mineAlpha(symbol string) <-chan *harnessResult {
+	return harnessCtl(symbol, "./mine-alpha 1")
+}
+
+// mineBeta will mine a single block on the beta node of the asset indicated
+// by symbol.
+func mineBeta(symbol string) <-chan *harnessResult {
+	return harnessCtl(symbol, "./mine-beta 1")
+}
+
+// harnessResult is the result of a harnessCtl command.
+type harnessResult struct {
+	err    error
+	output string
+	cmd    string
+}
+
+// String returns a result string for the harnessResult.
+func (res *harnessResult) String() string {
+	if res.err != nil {
+		return fmt.Sprintf("error running harnes command %q: %v", res.cmd, res.err)
+	}
+	return fmt.Sprintf("response from harness command %q: %s", res.cmd, res.output)
+}
+
+// harnessCtl will run the command from the harness-ctl directory for the
+// specified symbol.
+func harnessCtl(symbol, cmd string, args ...interface{}) <-chan *harnessResult {
+	cmd = fmt.Sprintf(cmd, args...)
+	harnessCtl := filepath.Join(dextestDir, symbol, "harness-ctl")
+	compoundCmd := fmt.Sprintf("cd %s; %s", harnessCtl, cmd)
+	c := make(chan *harnessResult)
+	go func() {
+		output, err := exec.CommandContext(ctx, "/bin/sh", "-c", compoundCmd).Output()
```

Comment:
I guess you did it like this with `"/bin/sh"` as the command because of the `cd` to harness ctl?  You can also set the Command.Dir like the following so you don't need /bin/sh:

```go
func harnessCtl(symbol, cmd string, args ...string) <-chan *harnessResult {
	dir := filepath.Join(dextestDir, symbol, "harness-ctl")
	c := make(chan *harnessResult)
	go func() {
		fullCmd := strings.Join(append([]string{cmd}, args...), " ")
		command := exec.CommandContext(ctx, cmd, args...)
		command.Dir = dir
		output, err := command.Output()
		if err != nil {
			var exitErr *exec.ExitError
			if errors.As(err, &exitErr) {
				log.Errorf("exec error (%s) %q: %v: %s", symbol, fullCmd, err, string(exitErr.Stderr))
			} else {
				log.Errorf("%s harnessCtl error running %q: %v", symbol, fullCmd, err)
			}
		}
		c <- &harnessResult{
			err:    err,
			output: strings.TrimSpace(string(output)),
			cmd:    fullCmd,
		}
	}()
	return c
}
```

Then you change all the calls to be like

```go
harnessCtl(btc, "./alpha", "sendtoaddress", betaAddrBTC, valString(btcPer))
```

instead of 

```go
harnessCtl(btc, "./alpha sendtoaddress %s %s", betaAddrBTC, valString(btcPer))
```